src/devices/SensorHub/SensorHub.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using UnitsNet;
+
+namespace Iot.Device.SensorHub
+{
+    /// <summary>
+    /// SensorHub
+    /// </summary>
+    public class SensorHub
+    {
+        /// <summary>
+        /// Default I2C bus id
+        /// </summary>
+        public const int DefaultI2cBusId = 1;
+
+        /// <summary>
+        /// Default I2C address
+        /// </summary>
+        public const int DefaultI2cAddress = 0x17;
+
+        private readonly I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Creates a SensorHub with the default bus and connection values
+        /// </summary>
+        /// <returns>SensorHub</returns>
+        public static SensorHub Create()
+        {
+            return new(I2cDevice.Create(new I2cConnectionSettings(DefaultI2cBusId, DefaultI2cAddress)));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SensorHub"/> class.
+        /// </summary>
+        public SensorHub(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+        }
+
+        /// <summary>
+        /// Try to read the temperature from the off board thermometer in the range -30 to 127 Celsius
+        /// </summary>
+        /// <param name="temperature">The temperature in Celsius if successful</param>
+        /// <returns></returns>
+        public bool TryReadOffBoardTemperature(out Temperature temperature)
+        {
+            temperature = Temperature.MaxValue;
+            var status = ReadRegister(Register.STATUS_REG);
+            if (!((status & 0x1) == 0x1) && !((status & 0x2) == 0x2))
```

Comment:
maybe an enum for the Register status would be better than magic numbers

Reply:
If it's the only usage that is fine as well but make sure to add comment what magic numbers mean. If it's used in more than one place then enum is preferred